lib/rubocop/cop/style/redundant_return.rb

```diff
@@ -165,6 +167,18 @@ def message(node)
             MSG
           end
         end
+
+        def allowed_methods
+          cop_config['AllowedMethods']&.map(&:to_sym) || []
+        end
+
+        def method_name_before_of(node)
+          node_place_index = node.parent.children.find_index(node)
+          before_node = node.parent.children[node_place_index - 1]
```

Comment:
This will lead to all kinds of problems because you cannot make assumptions about what the parent of the return node will be.

For example you would allow this `return`:
```ruby
def foo
  if bar
    allowed_method
  else
    return
  end
end
```

Reply:
Thank you.
I added `node.parent.type` check